crates/fuel-gas-price-algorithm/gas-price-analysis/src/optimisation.rs

```diff
@@ -11,13 +11,13 @@ fn da_pid_factors(size: usize) -> Vec<(i64, i64)> {
         .collect()
 }
 
-pub fn naive_optimisation(iterations: usize, size: usize) -> (SimulationResults, (i64, i64)) {
+pub fn naive_optimisation(simulator: Simulator, iterations: usize, da_recording_rate: usize) -> (SimulationResults, (i64, i64)) {
```

Comment:
```suggestion
pub fn naive_optimisation(simulator: &Simulator, iterations: usize, da_recording_rate: usize) -> (SimulationResults, (i64, i64)) {
```

can be a reference? acc to the signature: https://github.com/FuelLabs/fuel-core/pull/2157/files#diff-a5477e1747a3035f9dcb256499945866dd5643c5988a0e5534de0752652b4bdbR39